cmd/server/workflow/workflow.proto

```diff
@@ -0,0 +1,146 @@
+syntax = "proto3";
+option go_package = "github.com/argoproj/argo/cmd/server/workflow";
+
+
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "k8s.io/apimachinery/pkg/apis/meta/v1/generated.proto";
+import "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1/generated.proto";
+import "k8s.io/api/core/v1/generated.proto";
+
+// Workflow Service
+//
+// Workflow Service API performs CRUD actions against application resources
+package workflow;
+
+
+message WorkflowCreateRequest {
+    string namespace = 1;
+    github.com.argoproj.argo.pkg.apis.workflow.v1alpha1.Workflow workflow = 2;
+    string instanceID = 3;
+    bool serverDryRun = 4;
+    k8s.io.apimachinery.pkg.apis.meta.v1.CreateOptions createOptions = 5;
+}
+
+message WorkflowGetRequest {
+    string workflowName = 1;
+    string namespace = 2;
+    k8s.io.apimachinery.pkg.apis.meta.v1.GetOptions getOptions = 3;
+}
+
+message WorkflowListRequest {
+    string namespace = 1;
+    k8s.io.apimachinery.pkg.apis.meta.v1.ListOptions listOptions = 2;
+}
+
+message WorkflowUpdateRequest {
+    string workflowName = 1;
+    string namespace = 2;
+    bool memoized = 3;
```

Comment:
this field is only used by resubmission - but people were using it in the wrong requests - it's meaningless in other request, so we should delete this and replace it with:
* WorkflowResubmitRequest
* WorkflowSuspendRequest
* WorkflowResumeRequest
* etc

Reply:
@sarabala1979 
@simster7 watch out for this

Reply:
Agree with Alex. memoized should not be part of WorkflowUpdateRequest